oak-run/src/main/java/org/apache/jackrabbit/oak/run/GenerateFullGCCommand.java

```diff
@@ -0,0 +1,416 @@
+        long startGenTimestamp = System.currentTimeMillis();
+
+        List<String> generatedGarbageBasePaths = new ArrayList<>();
+
+        int numberOfRuns = options.getNumberOfRuns();
+        int intervalSeconds = options.getGenerateIntervalSeconds();
+        Runnable task = () -> {
+            if (continuousRunIndex < numberOfRuns) {
+                try {
+                    String genBasePath = generateGarbage(options, closer, continuousRunIndex, builder, startGenTimestamp);
+                    generatedGarbageBasePaths.add(genBasePath);
+                } catch (Exception e) {
+                    LOG.error("Error generating garbage in run " + continuousRunIndex, e);
+                }
+                LOG.info("Task executed. Count: " + (continuousRunIndex + 1));
+                continuousRunIndex++;
+            } else {
+                // Shutdown the executor once the task has run numberOfRuns times
+                executor.shutdown();
+                LOG.info("Task completed " + numberOfRuns + " times. Stopping execution.");
+            }
+        };
+
+        // Schedule the task to run every intervalSeconds
+        executor.scheduleAtFixedRate(task, 0, intervalSeconds, TimeUnit.SECONDS);
+
+        return generatedGarbageBasePaths;
+    }
+
+    /**
+     * Generate garbage nodes in the repository in order to allow for testing fullGC functionality.
+     *
+     * Returns the path of the generated FULLGC_GEN_BASE_PATH node (under the root).
+     * @param options
+     * @param closer
+     * @param runIndex
+     * @return
+     * @throws IOException
+     * @throws Exception
+     */
+    private String generateGarbage(GenerateFullGCOptions options, Closer closer, int runIndex) throws IOException, Exception {
+
+        DocumentNodeStoreBuilder<?> builder = createDocumentMKBuilder(options, closer);
+        long generationTimestamp = System.currentTimeMillis();
+
+        return generateGarbage(options, closer, runIndex, builder, generationTimestamp);
+    }
+
+    private String generateGarbage(GenerateFullGCOptions options, Closer closer, int runIndex,
+                                  DocumentNodeStoreBuilder<?> builder, long timestamp) throws IOException, Exception {
+
+        if (builder == null) {
+            System.err.println("generateFullGC mode only available for DocumentNodeStore");
+            System.exit(1);
+        }
+
+        String generationBasePath = FULLGC_GEN_BASE_PATH + timestamp + "_" + runIndex;
+        System.out.println("Generating fullGC on the document: " + generationBasePath);
+        documentNodeStore = builder.build();
+
+        VersionGarbageCollector.FullGCMode fullGCMode = getFullGCMode(options);
+        if (fullGCMode == VersionGarbageCollector.FullGCMode.NONE) {
+            LOG.error("Invalid garbageType specified. Must be one of the following: 1 - EMPTYPROPS, 2 - GAP_ORPHANS, 3 - GAP_ORPHANS_EMPTYPROPS");
```

Comment:
assume this log message will be outdated soon with more fullGC modes ready for rollout. eg we will support ALL_ORPHANS at some point etc... Perhaps just mention to use a valid fullGc mode (and perhaps add the classname)?

Reply:
Changed. Thank you for the suggestion.